### PR TITLE
docs: k8s + CoreOS + Cilium Stack building with Ansible

### PR DESCRIPTION
**Pull request by kubebc**
first pull

---
This change is [Reviewable](https://reviewable.io/reviews/cilium/cilium/1724)

### CODE REVIEW

**Comment by aalemayhu** [comment]
Thanks for sending the pull request. I tried rendering the page but looks like it's not linked properly so had to manually direct to it. The existing files have several examples, please compare them to your own and make appropriate changes.

**Comment by tgraf** [comment]
@kubebc This looks great, thanks a lot for doing this! Will have a look soon.

File: Documentation/guides/k8s-aws-ansible/install.rst [hunk 1]
```diff
@@ -0,0 +1,1430 @@
+*Kubernetes + CoreOS + Cilium Stack building with Ansible*
```
Comment (techcet) [inline Documentation/guides/k8s-aws-ansible/install.rst:1]: I suggest we link this into the Installation Guide, after the "Core OS" install section, before the "Installation from Source" section. It would be good to title it "Installation via Ansible" and then you can describe the environment  (Kubernetes + CoreOS + Cilium Stack building with Ansible) in an introductory sentence for this section, and include that the example is run in an AWS environment.

File: Documentation/guides/k8s-aws-ansible/install.rst [hunk 2]
```diff
@@ -0,0 +1,1430 @@
...
+::
+
+    192.168.192.10/19
+
+CILIUM 
+
+::
+
+    10.X.X.Y/8
+
```
Comment (techcet) [inline Documentation/guides/k8s-aws-ansible/install.rst:36]: For the above environment details, I suggest putting in tabular format to eliminate whitespace. Also a more descriptive subtitle section like "Environment IP Subnets" or "Environment Network Details".

File: Documentation/guides/k8s-aws-ansible/install.rst (hunk 1 above)
Comment (tgraf) [inline Documentation/guides/k8s-aws-ansible/install.rst:1]: The title sections should be formatted as follows
```
####
Title
####

***
Sub Title
***

Sub Sub title
=========

Sub Sub Sub title
-----------------
```

File: Documentation/guides/k8s-aws-ansible/install.rst [hunk 3]
```diff
@@ -0,0 +1,1430 @@
+*Kubernetes + CoreOS + Cilium Stack building with Ansible*
+
```
Comment (tgraf) [inline Documentation/guides/k8s-aws-ansible/install.rst:2]: Will be great to have a quick introduction section which describes the scope of the guide and what it achives

File: Documentation/guides/k8s-aws-ansible/install.rst [hunk 4]
```diff
@@ -0,0 +1,1430 @@
...
+IP SPACES
```
Comment (tgraf) [inline Documentation/guides/k8s-aws-ansible/install.rst:3]: Designate IP addressing space

File: Documentation/guides/k8s-aws-ansible/install.rst [hunk 5]
```diff
@@ -0,0 +1,1430 @@
...
+---------
+
```
Comment (tgraf) [inline Documentation/guides/k8s-aws-ansible/install.rst:5]: This guide uses the following IP addresses and subnets, please substitute based on your requirement and availability of IP addresses.

File: Documentation/guides/k8s-aws-ansible/install.rst [hunk 6]
```diff
@@ -0,0 +1,1430 @@
...
+    192.168.192.10/19
+
+CILIUM 
+
+::
+
+    10.X.X.Y/8
+
+1. Getting started with AWS
```
Comment (tgraf) [inline Documentation/guides/k8s-aws-ansible/install.rst:37]: Preparing the AWS environment

File: Documentation/guides/k8s-aws-ansible/install.rst [hunk 7]
```diff
@@ -0,0 +1,1430 @@
...
+----------------------------
+
+1.1 Create new vpc and network
```
Comment (tgraf) [inline Documentation/guides/k8s-aws-ansible/install.rst:40]: VPC

File: Documentation/guides/k8s-aws-ansible/install.rst [hunk 8]
```diff
@@ -0,0 +1,1430 @@
...
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
```
Comment (tgraf) [inline Documentation/guides/k8s-aws-ansible/install.rst:42]: Describe on which VM / host this needs to be run

File: Documentation/guides/k8s-aws-ansible/install.rst [hunk 9]
```diff
@@ -0,0 +1,1430 @@
...
+::
+
+    core@host:~/cilium$ mkdir installation;cd installation/
```
Comment (tgraf) [inline Documentation/guides/k8s-aws-ansible/install.rst:45]: I would make the prompt smaller, maybe just `~/cilium$ `

File: Documentation/guides/k8s-aws-ansible/install.rst [hunk 10]
```diff
@@ -0,0 +1,1430 @@
...
+
+::
+
+    core@host:~/cilium/installation$ export IMAGE_ID=ami-xxxxxxxx
+    core@host:~/cilium/installation$ aws ec2 describe-images --image-ids ami-xxxxxxxx
+
+
+1.3.1 CREATE MASTERS
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
```
Comment (tgraf) [inline Documentation/guides/k8s-aws-ansible/install.rst:85]: I assume that the `XXXXXXXXXXXXXXXXXXXXXXXXXXXX` needs to be substituted? Must be mentioned otherwise people will just copy & paste.

What is `${PRIVATE_IPV4}` set to?

File: Documentation/guides/k8s-aws-ansible/install.rst [hunk 11]
```diff
@@ -0,0 +1,1430 @@
...
+::
+
+    S1=`aws ec2 run-instances --image-id $IMAGE_ID --instance-type m4.large --key-name key --security-group-ids $SECURITY_GROUP_ID     --subnet $SUBNET_ID --private-ip-address 192.168.32.21 --block-device-mappings="[{\"DeviceName\":\"/dev/    xvda\",\"Ebs\":{\"DeleteOnTermination\":true,\"VolumeSize\":50,\"VolumeType\":\"gp2\"}}]" |grep InstanceId|awk -F'"' '{print $4}'`
+    
+    S2=`aws ec2 run-instances --image-id $IMAGE_ID --instance-type m4.large --key-name key --security-group-ids $SECURITY_GROUP_ID     --subnet $SUBNET_ID --private-ip-address 192.168.32.22 --block-device-mappings="[{\"DeviceName\":\"/dev/    xvda\",\"Ebs\":{\"DeleteOnTermination\":true,\"VolumeSize\":50,\"VolumeType\":\"gp2\"}}]" | grep InstanceId|awk -F'"' '{print $4}'`
+    
+    S3=`aws ec2 run-instances --image-id $IMAGE_ID --instance-type m4.large --key-name key --security-group-ids $SECURITY_GROUP_ID     --subnet $SUBNET_ID --private-ip-address 192.168.32.23 --block-device-mappings="[{\"DeviceName\":\"/dev/    xvda\",\"Ebs\":{\"DeleteOnTermination\":true,\"VolumeSize\":50,\"VolumeType\":\"gp2\"}}]" | grep InstanceId|awk -F'"' '{print $4}'`
+
+::
+
```
Comment (tgraf) [inline Documentation/guides/k8s-aws-ansible/install.rst:163]: Verifying the content of M1-M3 and S1-S3 seems to be done multiple times here

File: Documentation/guides/k8s-aws-ansible/install.rst (hunk 8 above)
Comment (tgraf) [reply Documentation/guides/k8s-aws-ansible/install.rst:42]: Describe what the commands will do and which variables need to be substituted

File: Documentation/guides/k8s-aws-ansible/install.rst [hunk 12]
```diff
@@ -0,0 +1,1430 @@
...
+    core@host:~/cilium/installation$ aws ec2 create-route-table --vpc-id $VPC_ID
+    core@host:~/cilium/installation$ export ROUTE_TABLE_ID=rtb-xxxxxxxx
+    core@host:~/cilium/installation$ aws ec2 associate-route-table --subnet-id $SUBNET_ID --route-table-id $ROUTE_TABLE_ID
+    core@host:~/cilium/installation$ aws ec2 create-route --route-table-id $ROUTE_TABLE_ID --destination-cidr-block 0.0.0.0/0     --gateway-id $GATEWAY_ID
+    core@host:~/cilium/installation$ aws ec2 modify-vpc-attribute --vpc-id=$VPC_ID --enable-dns-support
+    core@host:~/cilium/installation$ aws ec2 modify-vpc-attribute --vpc-id=$VPC_ID --enable-dns-hostnames
+
+1.2 Configuring Key Pair and Security Group 
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
```
Comment (tgraf) [inline Documentation/guides/k8s-aws-ansible/install.rst:63]: Needs 1-2 sentences describing what this will do. What does `SECURITY_GROUP_ID` need to be set to?

File: Documentation/guides/k8s-aws-ansible/install.rst [hunk 13]
```diff
@@ -0,0 +1,1430 @@
...
+                "contents": "[Unit]\nRequires=coreos-metadata.service\nAfter=coreos-metadata.service\n\n[Service]\nEnvironmentFile=/run/    metadata/coreos\nExecStart=\nExecStart=/usr/lib/coreos/etcd-wrapper $ETCD_OPTS \\\n      --listen-peer-urls=\"http://${COREOS_EC2_IPV4_LOCAL}:2380\" \\\n  --listen-client-urls=\"http://0.0.0.0:2379\" \\\n      --initial-advertise-peer-urls=\"http://${COREOS_EC2_IPV4_LOCAL}:2380\" \\\n      --advertise-client-urls=\"http://${COREOS_EC2_IPV4_LOCAL}:2379\" \\\n  --discovery=\"https://discovery.etcd.io/    XXXXXXXXXXXXXXXXXXXXXXXXXXXX\""
+              }
+            ]
+          }
+        ]
+      },
+      "networkd": {},
+      "passwd": {}
+    }        
+
```
Comment (tgraf) [inline Documentation/guides/k8s-aws-ansible/install.rst:140]: Mention that you are creating 3 masters here

File: Documentation/guides/k8s-aws-ansible/install.rst [hunk 14]
```diff
@@ -0,0 +1,1430 @@
...
+    
+    M2=`aws ec2 run-instances --image-id $IMAGE_ID --instance-type m3.medium --key-name key --security-group-ids $SECURITY_GROUP_ID     --subnet $SUBNET_ID --private-ip-address 192.168.32.12 --block-device-mappings="[{\"DeviceName\":\"/dev/    xvda\",\"Ebs\":{\"DeleteOnTermination\":true,\"VolumeSize\":50,\"VolumeType\":\"gp2\"}}]" --user-data file://master-config.json |grep     InstanceId|awk -F'"' '{print $4}'`
+    
+    M3=`aws ec2 run-instances --image-id $IMAGE_ID --instance-type m3.medium --key-name key --security-group-ids $SECURITY_GROUP_ID     --subnet $SUBNET_ID --private-ip-address 192.168.32.13 --block-device-mappings="[{\"DeviceName\":\"/dev/    xvda\",\"Ebs\":{\"DeleteOnTermination\":true,\"VolumeSize\":50,\"VolumeType\":\"gp2\"}}]" --user-data file://master-config.json |grep     InstanceId|awk -F'"' '{print $4}'`
+
+
+1.3.2 CREATE SLAVES 
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
```
Comment (tgraf) [inline Documentation/guides/k8s-aws-ansible/install.rst:153]: Mention that you are creating 3 slaves here

File: Documentation/guides/k8s-aws-ansible/install.rst [hunk 15]
```diff
@@ -0,0 +1,1430 @@
...
+    echo "export M2=i-xxxxxxxxxxxxxxxxxxx" >> gocilium.sh
+    echo "export M3=i-xxxxxxxxxxxxxxxxxxx" >> gocilium.sh
+    echo "export S1=i-xxxxxxxxxxxxxxxxxxx" >> gocilium.sh
+    echo "export S2=i-xxxxxxxxxxxxxxxxxxx" >> gocilium.sh
+    echo "export S3=i-xxxxxxxxxxxxxxxxxxx" >> gocilium.sh
+
+
+1.3.3 tag VMs 
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
```
Comment (tgraf) [inline Documentation/guides/k8s-aws-ansible/install.rst:206]: Mention the tags and the purpose of the tags

File: Documentation/guides/k8s-aws-ansible/install.rst [hunk 16]
```diff
@@ -0,0 +1,1430 @@
...
+    aws ec2 modify-instance-attribute --instance-id $S1 --source-dest-check "{\"Value\": false}"
+    
+    aws ec2 create-tags --resources $S2 --tags Key=Name,Value=ciliumSlave2
+    aws ec2 create-tags --resources $S2 --tags Key=role,Value=slave
+    aws ec2 modify-instance-attribute --instance-id $S2 --source-dest-check "{\"Value\": false}"
+    
+    aws ec2 create-tags --resources $S3 --tags Key=Name,Value=ciliumSlave3
+    aws ec2 create-tags --resources $S3 --tags Key=role,Value=slave
+    aws ec2 modify-instance-attribute --instance-id $S3 --source-dest-check "{\"Value\": false}"
+
```
Comment (tgraf) [inline Documentation/guides/k8s-aws-ansible/install.rst:236]: Mention that we now verify the setup with IPs and tags

File: Documentation/guides/k8s-aws-ansible/install.rst [hunk 17]
```diff
@@ -0,0 +1,1430 @@
...
+    i-xxxxxxxxxxxxxxxxxxx   192.168.32.23   x.x.x.x ciliumSlave3    ami-xxxxxxxx    running us-west-1a  2017-06-15T13:02:34.000Z
+
+
+
+2. Cluster TLS using OpenSSL
+----------------------------
+
+2.1 Kubernetes API Server Keypair
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
```
Comment (tgraf) [inline Documentation/guides/k8s-aws-ansible/install.rst:255]: Needs a description that we are generating the SSL key here

File: Documentation/guides/k8s-aws-ansible/install.rst [hunk 18]
```diff
@@ -0,0 +1,1430 @@
...
+    e is 65537 (0x10001)
+    core@host:~/cilium/installation/caNew$ openssl req -new -key apiserver-key.pem -out apiserver.csr -subj "/CN=kube-apiserver" -config openssl.cnf
+    core@host:~/cilium/installation/caNew$ openssl x509 -req -in apiserver.csr -CA ca.pem -CAkey ca-key.pem -CAcreateserial -out apiserver.pem -days 365 -extensions v3_req -extfile openssl.cnf
+    Signature ok
+    subject=/CN=kube-apiserver
+    Getting CA Private Key
+
+2.2 Kubernetes Worker Keypairs
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
```
Comment (tgraf) [inline Documentation/guides/k8s-aws-ansible/install.rst:293]: Needs a description that we are generating the key for the slaves here and what the key will be used for

File: Documentation/guides/k8s-aws-ansible/install.rst [hunk 19]
```diff
@@ -0,0 +1,1430 @@
...
+    aws ec2 describe-instances --query 'Reservations[*].Instances[*].[InstanceId,PrivateIpAddress,PublicIpAddress,Tags[0].Value,ImageId,State.Name,Placement.AvailabilityZone,LaunchTime]' --filters Name=tag:Name,Values=cilium* --output text |sort -k 4|grep -v None
+    i-xxxxxxxxxxxxxxxxxxx   192.168.32.11   x.x.x.x ciliumMaster1   ami-xxxxxxxx    running us-west-1a  2017-06-16T06:08:24.000Z
+    i-xxxxxxxxxxxxxxxxxxx   192.168.32.12   x.x.x.x ciliumMaster2   ami-xxxxxxxx    running us-west-1a  2017-06-16T06:08:24.000Z
+    i-xxxxxxxxxxxxxxxxxxx   192.168.32.13   x.x.x.x ciliumMaster3   ami-xxxxxxxx    running us-west-1a  2017-06-16T06:08:24.000Z
+    i-xxxxxxxxxxxxxxxxxxx   192.168.32.21   x.x.x.x ciliumSlave1    ami-xxxxxxxx    running us-west-1a  2017-06-16T06:08:24.000Z
+    i-xxxxxxxxxxxxxxxxxxx   192.168.32.22   x.x.x.x ciliumSlave2    ami-xxxxxxxx    running us-west-1a  2017-06-16T06:08:24.000Z
+    i-xxxxxxxxxxxxxxxxxxx   192.168.32.23   x.x.x.x ciliumSlave3    ami-xxxxxxxx    running us-west-1a  2017-06-16T06:08:24.000Z
+
+Backup current ones first
```
Comment (tgraf) [inline Documentation/guides/k8s-aws-ansible/install.rst:342]: Unclear what this does, just extracting the IPs of the master and slaves?

File: Documentation/guides/k8s-aws-ansible/install.rst [hunk 20]
```diff
@@ -0,0 +1,1430 @@
...
+
+    x.x.x.x     mp1
+    x.x.x.x     mp2
+    x.x.x.x     mp3
+    x.x.x.x     sp1
+    x.x.x.x     sp2
+    x.x.x.x     sp3
+
+| Update hosts file
+| $ sudo vi /etc/hosts
```
Comment (tgraf) [inline Documentation/guides/k8s-aws-ansible/install.rst:362]: This step seems unrelated with the copying of the keys. Why is this needed?

File: Documentation/guides/k8s-aws-ansible/install.rst [hunk 21]
```diff
@@ -0,0 +1,1430 @@
...
+    x.x.x.x     mp2 
+    x.x.x.x     mp3 
+    x.x.x.x     sp1 
+    x.x.x.x     sp2 
+    x.x.x.x     sp3
+
+::
+
+    core@host:~/cilium/installation/caNew$
+    for host in mp1 mp2 mp3; do scp ca.pem ca-key.pem apiserver.pem apiserver-key.pem core@${host}:~/; done
```
Comment (tgraf) [inline Documentation/guides/k8s-aws-ansible/install.rst:383]: This seems to be the only step required to copy the keys

File: Documentation/guides/k8s-aws-ansible/install.rst [hunk 22]
```diff
@@ -0,0 +1,1430 @@
...
+    x.x.x.x
+    [mp3]    
+    x.x.x.x
+    [sp1]    
+    x.x.x.x
+    [sp2]      
+    x.x.x.x
+    [sp3]     
+    x.x.x.x 
+    core@host:~/cilium/installation/ansible$ ansible-playbook -i hosts site.yml
```
Comment (tgraf) [inline Documentation/guides/k8s-aws-ansible/install.rst:443]: Where is site.yml coming from here?

File: Documentation/guides/k8s-aws-ansible/install.rst [hunk 23]
```diff
@@ -0,0 +1,1430 @@
...
+    
+    PLAY [mps,sps] **********************************************************************************************************************    ***************************************
+    
+    TASK [defunctzombie.coreos-bootstrap : Check if bootstrap is needed]     *******************************************************************************************************
```
Comment (tgraf) [inline Documentation/guides/k8s-aws-ansible/install.rst:447]: Can we shorten the output here, it's OK to mention that there will be output and the user only has to check for fatal errors.

File: Documentation/guides/k8s-aws-ansible/install.rst [hunk 24]
```diff
@@ -0,0 +1,1430 @@
...
+    
+    PLAY RECAP **************************************************************************************************************************    ***************************************
+    x.x.x.x               : ok=7    changed=6    unreachable=0    failed=0   
+    x.x.x.x                : ok=7    changed=6    unreachable=0    failed=0   
+    x.x.x.x              : ok=7    changed=6    unreachable=0    failed=0   
+    x.x.x.x                : ok=7    changed=6    unreachable=0    failed=0   
+    x.x.x.x             : ok=7    changed=6    unreachable=0    failed=0   
+    x.x.x.x                : ok=7    changed=6    unreachable=0    failed=0   
+    
+    core@host:~/cilium/installation/ansible$ ansible -i hosts mps,sps -m shell -a "free"
```
Comment (tgraf) [inline Documentation/guides/k8s-aws-ansible/install.rst:523]: Is this step strictly required?

File: Documentation/guides/k8s-aws-ansible/install.rst [hunk 25]
```diff
@@ -0,0 +1,1430 @@
...
+    -/+ buffers/cache:     197592    7980692
+    Swap:            0          0          0
+
+
+3.2 TLS Assets 
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+::
+
+    core@host:~/cilium/installation/ansible$
```
Comment (tgraf) [inline Documentation/guides/k8s-aws-ansible/install.rst:566]: Needs a description that this verifies whether the keys have been copied properly. The output can be truncated to a single node with a note that says the output should be the same for each node

File: Documentation/guides/k8s-aws-ansible/install.rst [hunk 26]
```diff
@@ -0,0 +1,1430 @@
...
+    
+    x.x.x.x | SUCCESS | rc=0 >>
+    apiserver-key.pem
+    apiserver.pem
+    bin
+    ca-key.pem
+    ca.pem
+    pypy
+    $ ansible -i hosts mps -b -m shell -a "mkdir -p /etc/kubernetes/ssl"
+    $ ansible -i hosts mps -b -m shell -a "mv apiserver-key.pem  apiserver.pem  ca-key.pem  ca.pem /etc/kubernetes/ssl/"
```
Comment (tgraf) [inline Documentation/guides/k8s-aws-ansible/install.rst:592]: Why did we not scp them do this location in the first place? The guide is getting quite long if we `scp && ls && mv && ls` instead of just `scp && ls`

File: Documentation/guides/k8s-aws-ansible/install.rst [hunk 27]
```diff
@@ -0,0 +1,1430 @@
...
+    -rw-------. 1 root root 1.7K Jun 16 08:03 /etc/kubernetes/ssl/ca-key.pem
+
+
+
+3.3 Start VMs if stopped
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+::
+
+
```
Comment (tgraf) [inline Documentation/guides/k8s-aws-ansible/install.rst:626]: Can they be down if we just copied to them?

File: Documentation/guides/k8s-aws-ansible/install.rst [hunk 28]
```diff
@@ -0,0 +1,1430 @@
...
+    core@host:~/cilium/installation/ansible$ for i in $M1 $M2 $M3; do aws ec2 start-instances --instance-ids $i; done;
+    core@host:~/cilium/installation/ansible$ for i in $M1 $M2 $M3; do aws ec2 describe-instance-status --instance-ids $i; done;
+
+Re-check Public IPs
+
```
Comment (tgraf) [inline Documentation/guides/k8s-aws-ansible/install.rst:631]: What does the user have to check here?

File: Documentation/guides/k8s-aws-ansible/install.rst [hunk 29]
```diff
@@ -0,0 +1,1430 @@
...
+https://quay.io/repository/coreos/hyperkube?tag=latest&tab=tags
+
+    core@host:~/cilium/installation/ansible$ /usr/bin/ansible --version
+    ansible 2.3.0.0
+      config file = /etc/ansible/ansible.cfg
+      configured module search path = Default w/o overrides
+      python version = 2.7.6 (default, Oct 26 2016, 20:30:19) [GCC 4.8.4]
+      
+
+    core@host:~/cilium/installation/ansible$ ansible -i hosts mp1 -b -m setup --tree /tmp/facts
```
Comment (tgraf) [inline Documentation/guides/k8s-aws-ansible/install.rst:655]: What is `/tmp/facts`?

File: Documentation/guides/k8s-aws-ansible/install.rst [hunk 30]
```diff
@@ -0,0 +1,1430 @@
...
+                                     Dload  Upload   Total   Spent    Left  Speed
+    100  1285  100  1285    0     0  25185      0 --:--:-- --:--:-- --:--:--  313k
+
+
+3.6 Set Up Cillium For Network Policy 
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+::
+
+    core@host:~/cilium/installation/ansible$ ansible -i hosts mp1 -b -m shell -a "kubectl create -f cilium-ds.yaml"
```
Comment (tgraf) [inline Documentation/guides/k8s-aws-ansible/install.rst:999]: Needs a description where `cilium-ds.yaml` is coming from

**Review by tgraf — CHANGES_REQUESTED** [review]
Great start, needs more description for each step and it would be great if we can truncated the output a bit in places.

Given that there are a lot of scripts. it will be great to use `literalinclude` of `RST` to include them from files so users can actually just download and modifying the scripts instead of using copy & paste each individual block.

File: Documentation/guides/k8s-aws-ansible/install.rst (hunk 9 above)
Comment (techcet) [inline Documentation/guides/k8s-aws-ansible/install.rst:45]: We still need to make a Style Guide for documentation, but in the mean time, can you please remove any text prior to the prompt '$' for all console output please?

File: Documentation/guides/k8s-aws-ansible/install.rst [hunk 31]
```diff
@@ -0,0 +1,1430 @@
...
+    i-xxxxxxxxxxxxxxxxxxx   192.168.32.12   x.x.x.x ciliumMaster2   ami-xxxxxxxx    running us-west-1a  2017-06-16T06:08:24.000Z
+    i-xxxxxxxxxxxxxxxxxxx   192.168.32.13   x.x.x.x ciliumMaster3   ami-xxxxxxxx    running us-west-1a  2017-06-16T06:08:24.000Z
+    i-xxxxxxxxxxxxxxxxxxx   192.168.32.21   x.x.x.x ciliumSlave1    ami-xxxxxxxx    running us-west-1a  2017-06-16T06:08:24.000Z
+    i-xxxxxxxxxxxxxxxxxxx   192.168.32.22   x.x.x.x ciliumSlave2    ami-xxxxxxxx    running us-west-1a  2017-06-16T06:08:24.000Z
+    i-xxxxxxxxxxxxxxxxxxx   192.168.32.23   x.x.x.x ciliumSlave3    ami-xxxxxxxx    running us-west-1a  2017-06-16T06:08:24.000Z
+
+Backup current ones first
+
+::
+
```
Comment (techcet) [inline Documentation/guides/k8s-aws-ansible/install.rst:345]: Is there a command missing below to provide the output?

File: Documentation/guides/k8s-aws-ansible/install.rst (hunk 20 above)
Comment (techcet) [inline Documentation/guides/k8s-aws-ansible/install.rst:362]: Should this line be in the code block below? Seems to be a formatting issue.

File: Documentation/guides/k8s-aws-ansible/install.rst [hunk 32]
```diff
@@ -0,0 +1,1430 @@
...
+    i-xxxxxxxxxxxxxxxxxxx   192.168.32.13   x.x.x.x ciliumMaster3   ami-xxxxxxxx    running us-west-1a  2017-06-29T08:43:03.000Z
+
+
+3.4 Create and run ansible playbook 
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+::
+
+Create the roles/masters
+https://quay.io/repository/coreos/hyperkube?tag=latest&tab=tags
```
Comment (techcet) [inline Documentation/guides/k8s-aws-ansible/install.rst:646]: Need more explanation in sentence on why pointing to the link with the Quay repository tags.

File: Documentation/guides/k8s-aws-ansible/install.rst [hunk 33]
```diff
@@ -0,0 +1,1430 @@
...
+
+    core@host:~/cilium/installation/ansible$ /usr/bin/ansible --version
```
Comment (techcet) [inline Documentation/guides/k8s-aws-ansible/install.rst:648]: The following output should be in a code block.

File: Documentation/guides/k8s-aws-ansible/install.rst [hunk 34]
```diff
@@ -0,0 +1,1430 @@
...
+    $ for i in $S1 $S2 $S3; do aws ec2 start-instances --instance-ids $i; done;
+    $ aws ec2 describe-instances --query 'Reservations[*].Instances[*].[InstanceId,PrivateIpAddress,PublicIpAddress,Tags[0].Value,ImageId,State.Name,Placement.AvailabilityZone,LaunchTime]' --filters Name=tag:Name,Values=cilium* --output text |sort -k 4|grep -v None
+    i-xxxxxxxxxxxxxxxxxxx   192.168.32.11   x.x.x.x ciliumMaster1   ami-xxxxxxxx    running us-west-1a  2017-06-29T08:43:00.000Z
+    i-xxxxxxxxxxxxxxxxxxx   192.168.32.12   x.x.x.x ciliumMaster2   ami-xxxxxxxx    running us-west-1a  2017-06-29T08:43:01.000Z
+    i-xxxxxxxxxxxxxxxxxxx   192.168.32.13   x.x.x.x ciliumMaster3   ami-xxxxxxxx    running us-west-1a  2017-06-29T08:43:03.000Z
+    i-xxxxxxxxxxxxxxxxxxx   192.168.32.21   x.x.x.x ciliumSlave1    ami-xxxxxxxx    running us-west-1a  2017-06-29T13:34:51.000Z
+    i-xxxxxxxxxxxxxxxxxxx   192.168.32.22   x.x.x.x ciliumSlave2    ami-xxxxxxxx    running us-west-1a  2017-06-29T13:34:52.000Z
+    i-xxxxxxxxxxxxxxxxxxx   192.168.32.23   x.x.x.x ciliumSlave3    ami-xxxxxxxx    running us-west-1a  2017-06-29T13:34:53.000Z
+
+update new ips here
```
Comment (techcet) [inline Documentation/guides/k8s-aws-ansible/install.rst:1071]: IPs

File: Documentation/guides/k8s-aws-ansible/install.rst [hunk 35]
```diff
@@ -0,0 +1,1430 @@
...
+::
+
+    core@host:~/cilium/installation/caNew$ env |grep kube
+    KUBECONFIG=/home/core/cilium/installation/.kube/config
+
+    core@host:~/cilium/installation$ cd caNew/
+    core@host:~/cilium/installation/caNew$ kubectl config set-cluster default-cluster --server=http://api.x.x.x.x:8080 --certificate-authority=ca.pem
+    Cluster "default-cluster" set.
+    core@host:~/cilium/installation/caNew$ kubectl config set-credentials default-admin --certificate-authority=ca.pem --client-key=admin-key.pem --client-certificate=admin.pem
+    User "default-admin
```
Comment (techcet) [inline Documentation/guides/k8s-aws-ansible/install.rst]: The following output should be in a code block.

File: Documentation/guides/k8s-aws-ansible/install.rst (hunk 3 above)
Comment (techcet) [reply Documentation/guides/k8s-aws-ansible/install.rst:2]: Please also state assumptions or minimum requirements to proceed (e.g. aws cli installed, AWS credentials set up, etc)

File: Documentation/guides/k8s-aws-ansible/install.rst [hunk 36]
```diff
@@ -0,0 +1,1430 @@
...
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+::
+
+    core@host:~/cilium/installation$ wget https://github.com/coreos/container-linux-config-transpiler/releases/download/v0.3.1/    ct-v0.3.1-x86_64-unknown-linux-gnu
+    core@host:~/cilium/installation$ chmod +x ct-v0.3.1-x86_64-unknown-linux-gnu
+
+::
+
+    core@host:~/cilium/installation$ vi master-config.yaml
```
Comment (techcet) [inline Documentation/guides/k8s-aws-ansible/install.rst:93]: Consider pulling the comments out of the code block below and using as an introductory statement to the file creation.

File: Documentation/guides/k8s-aws-ansible/install.rst [hunk 37]
```diff
@@ -0,0 +1,1430 @@
...
+    basicConstraints = CA:FALSE
+    keyUsage = nonRepudiation, digitalSignature, keyEncipherment
+    subjectAltName = @alt_names
+    [alt_names]
+    DNS.1 = kubernetes
+    DNS.2 = kubernetes.default
+    DNS.3 = kubernetes.default.svc
+    DNS.4 = kubernetes.default.svc.cluster.local
+    IP.1 = 192.168.192.1
+    IP.2 = x.x.x.x
```
Comment (techcet) [inline Documentation/guides/k8s-aws-ansible/install.rst:274]: Why is IP.2 the only one anonymized? Be consistent (all showing, or only an example and the rest anonymized).

File: Documentation/guides/k8s-aws-ansible/install.rst [hunk 38]
```diff
@@ -0,0 +1,1430 @@
...
+
+::
+
+    core@host:~/cilium/installation/caNew$
+    for host in mp1 mp2 mp3; do scp ca.pem ca-key.pem apiserver.pem apiserver-key.pem core@${host}:~/; done
+    scp ca.pem S1-worker.pem S1-worker-key.pem core@sp1:~/;
+    scp ca.pem S2-worker.pem S2-worker-key.pem core@sp2:~/;
+    scp ca.pem S3-worker.pem S3-worker-key.pem core@sp3:~/;
+
+3. Deploy Kubernetes Master Node(s)]
```
Comment (techcet) [inline Documentation/guides/k8s-aws-ansible/install.rst:388]: Extraneous ] at the end

File: Documentation/guides/k8s-aws-ansible/install.rst [hunk 39]
```diff
@@ -0,0 +1,1430 @@
...
+-----------------------------------
+
+3.1 Set up ANSIBLE
```
Comment (techcet) [inline Documentation/guides/k8s-aws-ansible/install.rst:391]: Ansible (not fully capitalized)

File: Documentation/guides/k8s-aws-ansible/install.rst [hunk 40]
```diff
@@ -0,0 +1,1430 @@
...
+    Swap:            0          0          0
+
+
+3.2 TLS Assets 
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+::
+
+    core@host:~/cilium/installation/ansible$ 
+    $ ansible -i hosts mps -m shell -a "ls"
```
Comment (techcet) [inline Documentation/guides/k8s-aws-ansible/install.rst:567]: Extra prompt line?

**Review by techcet — COMMENTED** [review]
Thanks for putting this together! I agree with Thomas - please watch for formatting styles of titles/subtitles. Also, some introductory text after titles/subtitles would help break up the code blocks and make for easier reading. I plan to try this out for more feedback, but in the meantime, included some minor formatting feedback.
